app/services/hyrax/admin_set_create_service.rb

```diff
@@ -9,35 +9,68 @@ module Hyrax
-    # @return [TrueClass]
-    # @see AdminSet
-    def self.create_default_admin_set(admin_set_id:, title:)
-      admin_set = AdminSet.new(id: admin_set_id, title: Array.wrap(title))
-      begin
-        new(admin_set: admin_set, creating_user: nil).create
-      rescue ActiveFedora::IllegalOperation
-        # It is possible that another thread created the AdminSet just before this method
-        # was called, so ActiveFedora will raise IllegalOperation. In this case we can safely
-        # ignore the error.
-        Rails.logger.error("AdminSet ID=#{AdminSet::DEFAULT_ID} may or may not have been created due to threading issues.")
+  class AdminSetCreateService # rubocop:disable Metrics/ClassLength
+    DEFAULT_ID = 'admin_set/default'
+    DEFAULT_TITLE = ['Default Admin Set'].freeze
+
+    class << self
+      # @api public
+      # Creates the default AdminSet and corresponding data
+      # @param admin_set_id [String] The default admin set ID
+      # @param title [Array<String>] The title of the default admin set
+      # @return [TrueClass]
+      # @see AdminSet
+      # @deprecated
+      def create_default_admin_set(admin_set_id: DEFAULT_ID, title: DEFAULT_TITLE)
+        Deprecation.warn("'##{__method__}' will be removed in Hyrax 4.0.  " \
+                         "Instead, use 'Hyrax::AdminSetCreateService.find_or_create_default_admin_set'.")
+        admin_set = AdminSet.new(id: admin_set_id, title: Array.wrap(title))
+        begin
+          new(admin_set: admin_set, creating_user: nil).create
+        rescue ActiveFedora::IllegalOperation
+          # It is possible that another thread created the AdminSet just before this method
+          # was called, so ActiveFedora will raise IllegalOperation. In this case we can safely
+          # ignore the error.
+          Rails.logger.error("AdminSet ID=#{AdminSet::DEFAULT_ID} may or may not have been created due to threading issues.")
+        end
       end
-    end
 
-    # @api public
-    # Creates a non-default AdminSet and corresponding data
-    # @param admin_set [AdminSet] the admin set to operate on
-    # @param creating_user [User] the user who created the admin set
-    # @return [TrueClass, FalseClass] true if it was successful
-    # @see AdminSet
-    # @raise [RuntimeError] if you attempt to create a default admin set via this mechanism
-    def self.call(admin_set:, creating_user:, **kwargs)
-      raise "Use .create_default_admin_set to create a default admin set" if admin_set.default_set?
-      new(admin_set: admin_set, creating_user: creating_user, **kwargs).create
+      # @api public
+      # Finds the default AdminSet if it exists; otherwise, creates it and corresponding data
+      # @return [Hyrax::AdministrativeSet] The default admin set.
+      # @see Hyrax::AdministrativeSet
+      # @raise [RuntimeError] if admin set cannot be persisted
+      def find_or_create_default_admin_set
```

Comment:
Replacement for `.create_default_admin_set` in this class and `AdminSet.find_or_create_default_admin_set`